### ISSUE TITLE
Failed to build on Fedora 28

### ISSUE THREAD

**Issue by GitatHub** [shadowsocks/shadowsocks-libev#2051]
### What version of shadowsocks-libev are you using?
N/A

### What operating system are you using?
Fedora 28 x64

### What did you do?
```
su -c 'dnf copr enable librehat/shadowsocks'
su -c 'dnf update'
su -c 'dnf install shadowsocks-libev'
```

### What did you expect to see?
Install shadowsocks-libev

### What did you see instead?

Last metadata expiration check: $TIME.
No match for argument: shadowsocks-libev
Error: Unable to find a match

### What is your config in detail (with all sensitive info masked)?
N/A

-----------------
I guess the rpm for fedora 27 could work around,but it said:

`ss-local: error while loading shared libraries: libsodium.so.18: cannot open shared object file: No such file or directory
`
and the lib has been indeed installed:

```
sudo yum install libsodium
Last metadata expiration check: $TIME.
Package libsodium-1.0.16-4.fc28.x86_64 is already installed, skipping.
Dependencies resolved.
Nothing to do.

```

**Comment by yanqiyu** [comment]
The reason for not working with the RPM for fc27 doesn't work is that that rpm is compiled with another version of **libsodium**.
Two solutions: 

1. Install the correct version of libsodium
2. Compile yourself and please check #2036 's solution to the compilation error (Useful for both building a rpm or building just for install)

**Comment by librehat** [comment]
There were errors in the released version 3.1.3 that prevents the build from happening.
```
BUILDSTDERR: server.c:912:47: error: '__builtin_snprintf' output may be truncated before the last format character [-Werror=format-truncation=]
BUILDSTDERR: server.c:912:47: error: '__builtin_snprintf' output may be truncated before the last format character [-Werror=format-truncation=]
```

Taken from the [buildlog](https://copr-be.cloud.fedoraproject.org/results/librehat/shadowsocks/fedora-28-x86_64/00724124-shadowsocks-libev/build.log.gz)

**Comment by librehat** [comment]
From the diff I can see, `server.c` hasn't gotten any changes around these lines, re-open the issue
